### PR TITLE
fix: RunCommand calling scripts with incorrect executable path

### PR DESCRIPTION
**Pull request by chdsbd**
Fixes GH-965

Calling `sys.argv` should run the same program as the currently running program. To make calling Poetry scripts through RunCommand match this behavior, we must set `sys.argv[0]` to be the full path of the executable.

This change makes the behavior of calling a script through `poetry run` the same as calling a script directly from the .venv/bin.

# Pull Request Check List

This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://poetry.eustace.io/docs/contributing/) at least once, it will save you unnecessary review cycles!

- [x] Added **tests** for changed code.
- [x] Updated **documentation** for changed code.

**Note**: If your Pull Request introduces a new feature or changes the current behavior, it should be based
on the `develop` branch. If it's a bug fix or only a documentation update, it should be based on the `master` branch.

If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing!

### CODE REVIEW

**Comment by chdsbd** [comment]
My test is hacky. Any tips on cleaning it up would be appreciated.

**Comment by chdsbd** [comment]
@sdispater This is ready for review

**Comment by chdsbd** [comment]
This is still something I'd like to fix in poetry. I'll fix the conflicts when I get a commitment that this will be merged.

File: poetry/console/commands/run.py [hunk 1]
```diff
@@ -20,6 +20,13 @@ def handle(self):
         return self.env.execute(*args)
 
     def run_script(self, script, args):
+        # Calling `sys.argv` should run the same program as the currently
+        # running program. To make calling Poetry scripts through RunCommand
+        # match this behavior, we must set `sys.argv[0]` to be the full path of
+        # the executable.
+        full_path_args = list(args)
+        full_path_args[0] = self.env._bin(full_path_args[0])
```
Comment (finswimmer) [inline poetry/console/commands/run.py]: Couldn't we simplify it with

```
args[0] = self.env._bin(args[0])
```
instead?

**Comment by arbaes** [comment]
Hello ! 

I just ran into this issue, is there still no plan to merge this ?

**Comment by Secrus** [comment]
@chdsbd are you still interested in bringing this to Poetry? if so, please merge the master branch to your code and resolve potential issues. This would allow for rereview.

**Comment by qexat** [comment]
Any updates? This would be very nice if this issue was fixed..

**Comment by wagnerluis1982** [comment]
If desired, I can work on resolve the conflicts and address the issues.

@chdsbd are you okay if I rebase your work on another branch? (I will keep the authorship of your commits)

**Comment by chdsbd** [comment]
@wagnerluis1982 Thank you. That would be great!

**Comment by wagnerluis1982** [comment]
@finswimmer could you close this PR in favor of #6737?

**Comment by radoering** [comment]
Superseded by #6737

**Pull request closed without merging**